### ISSUE TITLE
Weak oracle result staleness check in `staleCheck()`

### ISSUE THREAD

**Issue by kitty-the-kat** [groLabs/GSquared#18]
`staleCheck()` checks if the Chainlink oracle price data is stale. The staleness check only checks if the timestamp is from the last 24 hours, but a stricter check would also check if the roundId is stale.

#### Technical Details

`staleCheck()` only checks that the Chainlink price data is under 24 hours old. The staleness check does not consider whether the roundId data may be outdated. It is recommended to do both, as shown in other security report findings [here](https://consensys.net/diligence/audits/2021/09/fei-protocol-v2-phase-1/#chainlinkoraclewrapper---latestrounddata-might-return-stale-results) and [here](https://github.com/code-423n4/2021-05-fairside-findings/issues/70). Specifically, the DAI/USD oracle [updates more regularly than every 24 hours](https://data.chain.link/ethereum/mainnet/stablecoins/dai-usd). Considering that the Gro protocol has protections in place for stablecoins losing their peg, improving the Chainlink price staleness check is a crucial consideration.

#### Impact

Low. The staleness check could provide a false positive, say in the case that the price data is 23 hours old but is not from the most recent roundId.

#### Recommendation

Consider modifying `setStrategies()` to the following:
```solidity
	function staleCheck(uint256 _updatedAt, uint80 _roundId, uint80 _answeredInRound) internal view returns (bool) {
        return ((block.timestamp - _updatedAt >= STALE_CHECK) && (_answeredInRound == _roundId));
    }
```

**Comment by engn33r** [comment]
@kitty-the-kat correct, `staleCheck()` is only called with the following call flows and the external functions `usdToStable()` and `stableToUsd()` in src/oracles/RouterOracle.sol are not used by the protocol.
1. `usdToStable()` -> `getPriceFeed()` -> `staleCheck()`
2. `stableToUsd()` -> `getPriceFeed()` -> `staleCheck()`

This is where `staleCheck()` is located: https://github.com/groLabs/GSquared-foundry/blob/f1831bdb353d4a0b3a8937087e1663f73b75e905/src/oracles/RouterOracle.sol#L121
